kratos/utilities/variable_utils.cpp

```diff
@@ -486,4 +487,94 @@ void VariableUtils::AuxiliaryAtomicAdd(
         rSumValue[2] += rPrivateValue[2];
 }
 
+template <>
+ModelPart::ElementsContainerType& VariableUtils::GetContainer<ModelPart::ElementsContainerType>(ModelPart& rModelPart)
+{
+    return rModelPart.Elements();
+}
+
+template <>
+ModelPart::ConditionsContainerType& VariableUtils::GetContainer<ModelPart::ConditionsContainerType>(ModelPart& rModelPart)
+{
+    return rModelPart.Conditions();
+}
+
+template <class TDataType, class TContainerType, class TWeightDataType>
+void VariableUtils::DistributeVariable(
+    ModelPart& rModelPart,
+    const Variable<TDataType>& rVariable,
+    const Variable<TWeightDataType>& rWeightVariable,
+    const bool IsInverseWeightProvided)
+{
+    KRATOS_TRY
+
+    SetNonHistoricalVariableToZero(rVariable, rModelPart.Nodes());
+
+    auto& r_entities = GetContainer<TContainerType>(rModelPart);
+    const int number_of_conditions = r_entities.size();
+
+    const std::function<double(const Node<3>&)>& r_direct_method =
+        [rWeightVariable](const Node<3>& rNode) -> double {
+            return rNode.GetValue(rWeightVariable);
+        };
+
+    const std::function<double(const Node<3>&)>& r_inverse_method =
+        [rWeightVariable](const Node<3>& rNode) -> double {
+            return 1.0 / rNode.GetValue(rWeightVariable);
+        };
+
+    const std::function<double(const Node<3>&)>& r_weight_method =
+        (IsInverseWeightProvided) ? r_inverse_method : r_direct_method;
+
+#pragma omp parallel for
+    for (int i_condition = 0; i_condition < number_of_conditions; ++i_condition)
+    {
+        auto it_entity = r_entities.begin() + i_condition;
```

Comment:
```suggestion
        auto it_entity = r_entities.begin() + i_entity;
```
To avoid possible misunderstandings.